changelog.d/3912.misc

```diff
@@ -0,0 +1 @@
+Add a regression test for logging failed HTTP requests on Python 3.
```

Comment:
Isn't the important thing in this PR that we fix the logging, rather than adding a test?

Reply:
I fixed the logging in another PR (that caused the exception). The logging fix here is just a sneaky b'' showing up fix :)